byzcoin/contract_darc.go

```diff
@@ -83,7 +87,7 @@ func (c *contractSecureDarc) Spawn(rst ReadOnlyStateTrie, inst Instruction, coin
 	// If we got here this is a spawn:xxx in order to spawn
 	// a new instance of contract xxx, so do that.
 
-	cfact, found := c.s.GetContractConstructor(inst.Spawn.ContractID)
+	cfact, found := c.contracts.Search(inst.Spawn.ContractID)
```

Comment:
What if `SetRegistry` has not been called? Shouldn't you first check if `c.contracts == nil` ? (nothing prevents you from not using `GetContract()`)